pkg/event/target/kafka.go

```diff
@@ -90,9 +105,22 @@ func (target *KafkaTarget) Close() error {
 	return target.producer.Close()
 }
 
-// NewKafkaTarget - creates new Kafka target.
+// NewKafkaTarget - creates new Kafka target with auth credentials.
 func NewKafkaTarget(id string, args KafkaArgs) (*KafkaTarget, error) {
 	config := sarama.NewConfig()
+
+        config.Net.SASL.User = KafkaArgs.Net.SASL.User
+        config.Net.SASL.Password = KafkaArgs.Net.SASL.Password
+        config.Net.SASL.Handshake = KafkaArgs.Net.SASL.Handshake
+        config.Net.SASL.Enable = KafkaArgs.Net.SASL.Enable
+
+        config.Net.TLS.Enable = KafkaArgs.Net.TLS.Enable
+        tlsConfig := &tls.Config{
+                InsecureSkipVerify: KafkaArgs.Net.TLS.Config.InsecureSkipVerify,
+                ClientAuth: KafkaArgs.Net.TLS.Config.ClientAuth,
+        }
+        config.Net.TLS.Config = tlsConfig
```

Comment:
Any configuration change requires version bump and migration code from old to new configuration.  You would need to do that in https://github.com/minio/minio/blob/master/cmd/config-versions.go, https://github.com/minio/minio/blob/master/cmd/config-current.go and https://github.com/minio/minio/blob/master/cmd/config-migrate.go